File: datasets/lua-keep-code_davinci_001_temp_0.2/HumanEval_30_get_positive.lua
-- Return only positive numbers in the list.
-- >>> get_positive([-1, 2, -4, 5, 6])
-- [2, 5, 6]
-- >>> get_positive([5, 3, -5, 2, -3, 3, 9, 0, 123, 1, -10])
-- [5, 3, 2, 3, 9, 123, 1]
local function get_positive(l)
  local result = {}
  for _, v in ipairs(l) do
    if v > 0 then
      table.insert(result, v)
    end
  end
  return result
end

lu = require('luaunit')

function test_humaneval()
local candidate = get_positive
    lu.assertEquals(candidate({-1, -2, 4, 5, 6}), {4, 5, 6})
    lu.assertEquals(candidate({5, 3, -5, 2, 3, 3, 9, 0, 123, 1, -10}), {5, 3, 2, 3, 3, 9, 123, 1})
    lu.assertEquals(candidate({-1, -2}), {})
    lu.assertEquals(candidate({}), {})
end

os.exit(lu.LuaUnit.run())

File: datasets/R-keep-code_davinci_001_temp_0.2/HumanEval_30_get_positive.R
# Return only positive numbers in the list.
# >>> get_positive([-1, 2, -4, 5, 6])
# [2, 5, 6]
# >>> get_positive([5, 3, -5, 2, -3, 3, 9, 0, 123, 1, -10])
# [5, 3, 2, 3, 9, 123, 1]
get_positive <- function(l) {
    l[l > 0]
}
get_positive([-1, 2, -4, 5, 6])
get_positive([5, 3, -5, 2, -3, 3, 9, 0, 123, 1, -10])


test_humaneval <- function() {
candidate <- get_positive
    if(!identical(candidate(c(-1, -2, 4, 5, 6)), c(4, 5, 6))){quit('no', 1)}
    if(!identical(candidate(c(5, 3, -5, 2, 3, 3, 9, 0, 123, 1, -10)), c(5, 3, 2, 3, 3, 9, 123, 1))){quit('no', 1)}
    if(!identical(candidate(c(-1, -2)), list())){quit('no', 1)}
    if(!identical(candidate(list()), list())){quit('no', 1)}
}
test_humaneval()